src/hapi/pipelines/database/idps.py

```diff
@@ -0,0 +1,97 @@
+"""Functions specific to the refugees theme."""
+
+from logging import getLogger
+from typing import Dict
+
+from hapi_schema.db_idps import DBIDPs
+from sqlalchemy.orm import Session
+
+from ..utilities.logging_helpers import add_message
+from . import admins
+from .base_uploader import BaseUploader
+from .metadata import Metadata
+
+logger = getLogger(__name__)
+
+
+class IDPs(BaseUploader):
+    def __init__(
+        self,
+        session: Session,
+        metadata: Metadata,
+        admins: admins.Admins,
+        results: Dict,
+    ):
+        super().__init__(session)
+        self._metadata = metadata
+        self._admins = admins
+        self._results = results
+
+    def populate(self) -> None:
+        logger.info("Populating IDPs table")
+        errors = set()
+        # self._results is a dictionary where the keys are the HDX dataset ID and the
+        # values are a dictionary with keys containing HDX metadata plus a "results" key
+        # containing the results, stored in a dictionary with admin levels as keys.
+        # There is only one dataset for now in the results dictionary, take first value
+        # (popitem returns a tuple with (key, value) so take the value)
+        dataset = self._results.popitem()[1]
+        dataset_name = dataset["hdx_stub"]
+        for admin_level, admin_results in dataset["results"].items():
+            # admin_results contains the keys "headers", "values", and "hapi_resource_metadata".
+            # admin_results["values"] is a list of dictionaries of the format:
+            # [{AFG: [1, 2], BFA: [3, 4]}, {AFG: [A, B], BFA: [C, D]} etc
+            # So the way to get info from it is values[i_hdx_key][pcode][i] where
+            # i is just an iterator for the number of rows for that particular p-code
+            resource_id = admin_results["hapi_resource_metadata"]["hdx_id"]
+            hxl_tags = admin_results["headers"][1]
+            values = admin_results["values"]
+            admin_codes = values[0].keys()
+            for admin_code in admin_codes:
```

Comment:
I was wondering why you didn't use the configurable scraper for HNO, and now that makes so much sense. Indeed there are a lot of inefficiencies in my DTM implementation, but let's just close our eyes and get it out the door. If we end up moving forward with HAPI / standardization then it will get refactored to the DTM scraper I'd imagine.